Mono.Cecil/SecurityDeclaration.cs

```diff
@@ -144,16 +158,22 @@ public SecurityDeclaration (SecurityAction action, byte [] blob)
 			if (!HasImage || signature == 0)
 				throw new NotSupportedException ();
 
-			return blob = module.Read (this, (declaration, reader) => reader.ReadSecurityDeclarationBlob (declaration.signature));
+			return module.Read (ref blob, this, (declaration, reader) => reader.ReadSecurityDeclarationBlob (declaration.signature));
 		}
 
 		void Resolve ()
 		{
 			if (resolved || !HasImage)
 				return;
 
-			module.Read (this, (declaration, reader) => reader.ReadSecurityDeclarationSignature (declaration));
-			resolved = true;
+			lock (module.SyncRoot) {
+
+				if (resolved)
+					return;
+
+                module.Read (this, (declaration, reader) => reader.ReadSecurityDeclarationSignature (declaration));
```

Comment:
Nit: indent looks wrong (and uses spaces instead of tabs)

Reply:
updated